### PR TITLE
Implemented evaluation context as defined in 0.2.0 spec

### PR DESCRIPTION
**Pull request by skyerus**

### CODE REVIEW

**Review by james-milligan — APPROVED** [review]

**Review by beeme1mr — APPROVED** [review]

File: pkg/openfeature/client.go [hunk 1]
```diff
@@ -9,6 +9,8 @@ import (
 type IClient interface {
 	Metadata() ClientMetadata
 	AddHooks(hooks ...Hook)
+	SetEvaluationContext(evalCtx EvaluationContext)
+	EvaluationContext() EvaluationContext
```
Comment (toddbaert) [inline pkg/openfeature/client.go:13]: Some other SDKs don't allow the setting of evaluation context directly on the client, but instead take it as a param to the `getClient` method on the top-level API.

Up to you which you think is better for the go SDK.

File: pkg/openfeature/client.go [hunk 2]
```diff
@@ -55,6 +59,16 @@ func (c *Client) AddHooks(hooks ...Hook) {
 	c.hooks = append(c.hooks, hooks...)
 }
 
+// SetEvaluationContext sets the client's evaluation context
+func (c *Client) SetEvaluationContext(evalCtx EvaluationContext) {
+	c.evaluationContext = evalCtx
+}
```
Comment (toddbaert) [inline pkg/openfeature/client.go:65]: see https://github.com/open-feature/golang-sdk/pull/42/files#r947913435

**Review by toddbaert — APPROVED** [review]

File: pkg/openfeature/client.go (hunk 1 above)
Comment (skyerus) [reply pkg/openfeature/client.go:13]: I think this allows for more configurability and it keeps the client constructor terse. Thoughts @james-milligan ?

Comment (james-milligan) [reply pkg/openfeature/client.go:13]: I do nto have any strong opinions here, however im leaning towards @skyerus's changes, it keeps things declarative and prevents a user needing to pass in an empty `EvaluationContext` into `getClient` when one is not needed.

**Pull request merged**